### ISSUE TITLE
How to call step() on a state from breakpoint?

### ISSUE THREAD

**Issue by beraphin** [angr/angr#837]
Here is my test code:
```
def test(state):
    state.step()

p=angr.Project(binary)
state=p.factory.entry_state()
state.inspect.b('statement',statement=0x4000000b2e,when=angr.BP_BEFORE,action=test)
simgr=p.factory.simgr(state)
simgr.run()
```
I wrote a program to filter some states, and these state were saved from breakpoints, and I wanna run these states again. This why I wrote the test code above. However I got an error after running it:
```
  File "test.py", line 10, in test
    state.step()
  File "/home/angr/angr-dev/angr/angr/sim_state.py", line 446, in step
    return self.project.factory.successors(self, **kwargs)
  File "/home/angr/angr-dev/angr/angr/factory.py", line 85, in successors
    if engine.check(state, inline=inline, **kwargs):
  File "/home/angr/angr-dev/angr/angr/engines/engine.py", line 85, in check
    r = self._check(state, *args, **kwargs)
  File "/home/angr/angr-dev/angr/angr/engines/failure.py", line 15, in _check
    if jumpkind in ('Ijk_EmFail', 'Ijk_MapFail') or jumpkind.startswith('Ijk_Sig'):
AttributeError: 'NoneType' object has no attribute 'startswith'
```
How can I fix it or is there another way to do this?
Much obliged

**Comment by rhelmot** [comment]
SimInspect breakpoints aren't breakpoints in the debugger sense - they're just instrumentation hooks. You can't step from them, since you're already in the middle of stepping. If you want to instrument control flow you need to do it from the simulation manager, either with the built-in tools for stepping or by writing an exploration technique.

**Comment by beraphin** [comment]
@rhelmot  Thanks for reply.
I guess I understand the reason why it doesn't work, but I still confused about the solutions. 
Using the state created from explore() does work, but in my program it is hard to modify the existing code. It means I can just handle with the states from breakpoints.
So I tried to do it from simulation manager, then I modified the code:
```
def test(state):
    global p
    simgr=p.factory.simgr(state)
    simgr.step()
    embed()

p=angr.Project(binary)
state=p.factory.entry_state()
state.inspect.b('statement',statement=0x4000000b2e,when=angr.BP_BEFORE,action=test)
simgr=p.factory.simgr(state)
simgr.run()
```
Unfortunately I still got the same result. I guess it's because the way I create the simgr adding the state to simgr's active stash directly, and step() is using the active state, so it's not different from calling state.step().
Is there anything wrong with the way I create the simgr? Or maybe it won't work with the "middle state" at all, is there way to get the parent state which is not in the middle?
